extension/src/vscode/external.ts

```diff
@@ -1,3 +1,21 @@
-import { env, Uri } from 'vscode'
+import { env, ProviderResult, Uri, window } from 'vscode'
 
 export const openUrl = (url: string) => env.openExternal(Uri.parse(url))
+
+export const getCallBackUrl = async (path: string) => {
+  const uri = await env.asExternalUri(
+    Uri.parse(`${env.uriScheme}://iterative.dvc${path}`)
+  )
+
+  return uri.toString()
+}
+
+export const waitForUriResponse = (path: string, onResponse: () => unknown) => {
+  window.registerUriHandler({
+    handleUri(uri: Uri): ProviderResult<void> {
+      if (uri.path.startsWith(path)) {
```

Comment:
For now, I'm using `uri.path.startsWith`, but we can change it to an `===` check once Studio is updated.

Reply:
`decodeURI()` does not work here?

Reply:
Yes, technically we could use `decodeURI` here to check the path, but we need the user to be sent to the vscode URL with the `windowId` query since I believe it helps VSCode to route the user correctly depending on certain circumstances. 

We could use `decodeURI` as a temporary solution though instead of `.startsWith()` though!